lib/puma/server.rb

```diff
@@ -598,6 +605,13 @@ def notify_safely(message)
     end
     private :notify_safely
 
+    def select_timeout
+      return 0 if shutting_down?
+
+      @idle_timeout if @ios_count > 0
```

Comment:
I don't understand that `@ios_count`.

It only ever increment right? So this means you ignore `idle_timeout` on the very first `select`, why?

Reply:
Because of this case:

https://github.com/joshuay03/puma/blob/9c4a15a158b554f67e59f33378eee10abeec8c14/test/test_puma_server.rb#L632-L644

More context in the last part of my comment here: https://github.com/puma/puma/pull/3209#discussion_r1311532499.

Reply:
I'm not married to this approach, but I think it does make sense that the timeout isn't set for the very first request? I was mostly going off of the description in the feature request: https://github.com/puma/puma/issues/2580.

Reply:
Ah I see. IMO it's not necessary, but that's more for Nate to decide.

But if you do keep it, since it's used as a boolean flag, you might as well make it a boolean and flip it with `@flag ||= true`

Reply:
Ya good point, I just assumed that having the count would come in handy later on, I'll keep this in mind though.
Will wait for Nate's input first.

Reply:
I think I understand what's going on. Is this effectively what you are saying?

```suggestion
      @idle_timeout if @has_ever_processed_a_request
```

Reply:
I don't think this check is necessary.

As the original issue said, we already have the feature of socket activation. So, presumably, people that use idle_timeout will use socket activation to deal with starting a new puma process to handle their requests. 

The point of socket activation with the idle_timeout feature is to reduce resource consumption. IMO that doesn't really jive with keeping a process alive, idling, doing nothing, until the first request comes in.

We can assume that anyone using this feature uses socket activation, so doesn't need this behavior.

Reply:
Resolved with 01f9c13d007f31020400b726c6099e72e691548b.